src/Compilers/CSharp/Portable/Syntax/ForStatementSyntax.cs

```diff
@@ -22,7 +22,25 @@ public partial class SyntaxFactory
         /// <summary>Creates a new ForStatementSyntax instance.</summary>
         public static ForStatementSyntax ForStatement(SyntaxToken forKeyword, SyntaxToken openParenToken, VariableDeclarationSyntax declaration, SeparatedSyntaxList<ExpressionSyntax> initializers, SyntaxToken firstSemicolonToken, ExpressionSyntax condition, SyntaxToken secondSemicolonToken, SeparatedSyntaxList<ExpressionSyntax> incrementors, SyntaxToken closeParenToken, StatementSyntax statement)
         {
-            return ForStatement(forKeyword, openParenToken, default(SyntaxToken), declaration, initializers, firstSemicolonToken, condition, secondSemicolonToken, incrementors, closeParenToken, statement);
+            return ForStatement(forKeyword, openParenToken, default(SyntaxToken), null, declaration, initializers, firstSemicolonToken, condition, secondSemicolonToken, incrementors, closeParenToken, statement);
+        }
+
+        /// <summary>Creates a new ForStatementSyntax instance.</summary>
+        public static ForStatementSyntax ForStatement(VariableDeclarationSyntax declaration, SeparatedSyntaxList<ExpressionSyntax> initializers, ExpressionSyntax condition, SeparatedSyntaxList<ExpressionSyntax> incrementors, StatementSyntax statement)
+        {
+            return ForStatement(
+                SyntaxFactory.Token(SyntaxKind.ForKeyword),
+                SyntaxFactory.Token(SyntaxKind.OpenParenToken),
+                default(SyntaxToken),
+                null,
```

Comment:
I just removed one item from this list, but sure, I'll do that.

---
In reply to: [73614851](https://github.com/dotnet/roslyn/pull/12917#discussion_r73614851) [](ancestors = 73614851)